services:
  app:
    build:
      context: .
    ports:
      - 3030:3030
    volumes:
      - .:/app
    networks:
      - cache-network
    environment:
      - VIRTUAL_HOST=my.home.local
      - VIRTUAL_PORT=3030
    restart: on-failure
    command: >
      sh -c "python main.py"

  proxy:
    image: jwilder/nginx-proxy
    container_name: proxy
    ports:
      - 80:80
      - 443:443
    environment:
      - DEFAULT_HOST=my.home.local
    volumes:
      - /var/run/docker.sock:/tmp/docker.sock:ro
      - ./certs:/etc/nginx/certs
    networks:
      - cache-network
    depends_on:
      - app

  db:
    image: postgres:15.8-bullseye
    restart: always
    environment:
      - POSTGRES_USER=ghost
      - POSTGRES_PASSWORD=rider
    volumes:
      - pgdata:/var/lib/postgresql/data
    networks:
      - cache-network
    ports:
      - "5432:5432"

networks:
  cache-network:
    driver: bridge

volumes:
  pgdata: